apps/protocol-frontend/src/hooks/useContributionCount.ts

```diff
@@ -0,0 +1,57 @@
+import { useQuery } from '@tanstack/react-query';
+import { useUser } from '../contexts/UserContext';
+import { subWeeks } from 'date-fns';
+
+type ContributionCountArgs = {
+  startDate: Date;
+  endDate: Date;
+  guildIds?: number[] | null;
+  excludeUnassigned?: boolean;
+};
+
+const useContributionCountInRange = (args: ContributionCountArgs) => {
+  const { userData, govrnProtocol: govrn } = useUser();
+
+  const { isLoading, isFetching, isError, error, data } = useQuery(
+    ['ContributionGetCount', args],
+    async () => {
+      if (!userData?.id) {
+        console.error('getUserContributionsCount has no userData.id');
+        return [];
+      }
+
+      return await govrn.custom.getContributionCountByDateForUserInRange({
+        id: userData?.id,
+        ...args,
+      });
+    },
+  );
+
+  return { isLoading, isError, isFetching, error, data };
+};
+
+export const useContributionCountInYear = (
```

Comment:
- if they are combined, they would be associated with the same key; library would treat them as the same request, with the same response and cache. 
- adding dynamic keys would somehow be buggy. 


I will try more testing to that since args are different l, this supposedly means that they have different keys!!

Reply:
Yup, it'll take the args into consideration for the keys. Good to test though!